### ISSUE TITLE
Exception thrown after successful COPY with message sequence containing more than one message.

### ISSUE THREAD

**Issue by AlexEkholm** [kisli/vmime#140]
I'm working on an application that involves fetching messages from a cyrus imap v2.5.9 server. After fetching a set of messages they are copied to another folder on the mailserver, but VMIME throws a "received invalid response" exception after the apparently successful COPY operation.

Note that this error does not occur when attempting to COPY a single message -- only when attempting to copy two or more messages.

Here is a snippet from the log.

```
2016-08-29T12:15:35.498697   mail-fetch   debug [tracer][imap:1] S: * LIST (\HasNoChildren) "." INBOX.201608
2016-08-29T12:15:35.498876   mail-fetch   debug [tracer][imap:1] S: a004 OK Completed (0.359 secs 2 calls)
2016-08-29T12:15:35.498952   mail-fetch   debug Copying 5 message(s) to folder: 201608
2016-08-29T12:15:35.499118   mail-fetch   debug [tracer][imap:2] C: a034 UID COPY 706921:706925 INBOX.201608
2016-08-29T12:15:36.315680   mail-fetch   debug [tracer][imap:2] S: * 8064 FETCH (FLAGS (\Seen) UID 721758)
2016-08-29T12:15:36.316020   mail-fetch   debug [tracer][imap:2] S: a034 OK [COPYUID 1472135626 706921:706925 26306:26310] Completed
2016-08-29T12:15:36.317290   mail-fetch   error Failed to fetch messages from mail server!
[...]: Throw in [...]
	Caused by: 
[...]: Throw in function [...]
Dynamic exception type: N5boost16exception_detail10clone_implIN4util3log16MessageExceptionEEE
std::exception::what: Failed to fetch mail messages!
	Caused by:
[...]: Throw in function [...]
Dynamic exception type: N5boost16exception_detail10clone_implIN4util3log16MessageExceptionEEE
std::exception::what: Failed to move mail message(s)!
	Caused by:
[...]: Throw in function [...]
Dynamic exception type: N5boost16exception_detail10clone_implIN4util3log16MessageExceptionEEE
std::exception::what: Failed to copy message(s) to folder: 201608
	Caused by:
Dynamic exception type: N5vmime10exceptions16invalid_responseE
std::exception::what: Received invalid response.
2016-08-29T12:15:36.317743   mail-fetch   debug Disconnecting from mail server....
2016-08-29T12:15:36.317907   mail-fetch   debug [tracer][imap:2] C: a035 LOGOUT
2016-08-29T12:15:36.318018   mail-fetch   debug [tracer][imap:2] C: Disconnecting
2016-08-29T12:15:36.318244   mail-fetch   debug [tracer][imap:1] C: a005 LOGOUT
2016-08-29T12:15:36.318399   mail-fetch   debug [tracer][imap:1] C: Disconnecting
```

**Comment by vincent-richard** [comment]
Hello!

Thank you for reporting this bug. I have fixed it with c7b0676; it was caused by the parser waiting for an invalid separator char in "uid-range" (< , > instead of < : >)

Vincent

**Comment by AlexEkholm** [comment]
Thank you for the quick response!

I can confirm that this issue is resolved in  [`c7b0676`](https://github.com/kisli/vmime/commit/c7b067661985ba76c95500b92f53600938a07d11).

It looks like you may have left some debug output enabled though:

```

Read line:
* LIST (\Noselect) "." ""


  ENTER(response), pos=0
> response 
* LIST (\Noselect) "." ""

[...]

                         ^
          ENTER(SPACE), pos=25
> response > continue_req_or_response_data > response_data > CRLF > SPACE 
* LIST (\Noselect) "." ""


                         ^
          LEAVE(SPACE), result=FALSE, pos=25
        LEAVE(CRLF), result=TRUE, pos=25
      LEAVE(response_data), result=TRUE, pos=0
    LEAVE(continue_req_or_response_data), result=TRUE, pos=0

Read line:
a002 OK Completed (0.000 secs)

**Comment by vincent-richard** [comment]
Oops, thank you for noticing. I have fixed it.
